src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.TypeInit.cs

```diff
@@ -15,6 +15,13 @@ namespace ILCompiler
     // Manages policies around static constructors (.cctors) and static data initialization.
     partial class CompilerTypeSystemContext
     {
+        private ExplicitClassConstructors _explicitClassConstructors;
```

Comment:
This leaking into the type system context is really unfortunate.

If we step away, the problem ordered eager class constructors are trying to solve is really only about trying to run a piece of code in an ordered fashion. A static constructor is just a mechanism.

Can we just make this an "Ordered list of methods to run" (i.e. fix the class library so that these types no longer have .cctors under `#if CORERT` in the first place and just have an `Initialize` method we run).

The Project N IL2IL toolchain actually renames the .cctor method on these so that the rest of the system doesn't need to care. Having it as a .cctor is really very awkward.